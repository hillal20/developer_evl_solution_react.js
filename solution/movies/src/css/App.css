.App {
  background: lightcoral;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  align-items: center;
}
.App .nav-bar {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.App .nav-bar button {
  width: 20%;
  height: 40px;
  margin: 5px;
  font-weight: bold;
  font-size: 20px;
}
.App .nav-bar button:hover {
  background: darkgrey;
}
.App .nav-bar input {
  width: 20%;
  height: 40px;
  margin: 5px;
  font-weight: bold;
  font-size: 20px;
}
.App .all-movies {
  display: flex;
  flex-wrap: wrap;
}
.App .all-movies .movie-card {
  background: khaki;
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
  width: 30%;
  height: 30%;
  border: 3px solid red;
  margin: 10px;
  align-items: center;
  padding: 10px;
  border-radius: 20px;
}
.App .movie-detail {
  width: 30%;
  background: khaki;
  padding-bottom: 5%;
  border: 4px solid red;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 5%;
}
.App .movie-detail img {
  margin: 15px;
}
