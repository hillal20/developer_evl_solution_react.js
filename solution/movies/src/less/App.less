//less-watch-compiler  less css App.less

.App {
  background: lightcoral;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  align-items: center;
  .nav-bar {
    display: flex;
    flex-direction: column;
    align-items: center;
    button {
      width: 20%;
      height: 40px;
      margin: 5px;
      font-weight: bold;
      font-size: 20px;
      &:hover {
        background: darkgrey;
      }
    }
    input {
      width: 20%;
      height: 40px;
      margin: 5px;
      font-weight: bold;
      font-size: 20px;
    }
  }
  .all-movies {
    display: flex;
    flex-wrap: wrap;
    .movie-card {
      background: khaki;
      display: flex;
      flex-direction: column;
      margin-bottom: 10px;
      width: 30%;
      height: 30%;
      border: 3px solid red;
      margin: 10px;
      align-items: center;
      padding: 10px;
      border-radius: 20px;
    }
  }
  .movie-detail {
    width: 30%;
    background: khaki;
    padding-bottom: 5%;
    border: 4px solid red;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 5%;
    img {
      margin: 15px;
    }
  }
}
