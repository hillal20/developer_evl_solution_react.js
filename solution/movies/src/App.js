import React, { Component } from "react";
import "./css/App.css";
import AllMovies from "./allmovies/allmovies";
import MovieDetail from "./movieDetail/movieDetail";
import { BrowserRouter as Router, Route } from "react-router-dom";
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Router>
        <div className="App">
          <Route exact path="/" component={AllMovies} />
          <Route exact path="/detail/:id" component={MovieDetail} />
        </div>
      </Router>
    );
  }
}

export default App;
