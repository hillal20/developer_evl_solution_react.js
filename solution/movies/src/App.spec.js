import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import sinon from "sinon";
import Enzyme, { shallow, configure } from "enzyme";
import Renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("<app/>", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
  });

  it("create  snapshot", () => {
    let element = Renderer.create(<App />);
    expect(element.toJSON()).toMatchSnapshot();
  });

  it("should not defer from snapshot", () => {
    let element = shallow(<App />);
    expect(element).toMatchSnapshot();
  });

  it("should state.total is NaN when component mount first time", () => {
    const expected = NaN;
    const component = shallow(<App />);
    const instance = component.instance();
    expect(parseInt(instance.state.total)).toEqual(expected);
  });
});
