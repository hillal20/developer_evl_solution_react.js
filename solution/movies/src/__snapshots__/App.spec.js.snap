// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> create  snapshot 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="nav-bar"
    >
      <h1>
         Welcome to Movies guide 
      </h1>
      <button
        onClick={[Function]}
      >
        All Movies
      </button>
      <button
        onClick={[Function]}
      >
        Top Movies
      </button>
      <button
        onClick={[Function]}
      >
        Playing Movies
      </button>
      <button
        onClick={[Function]}
      >
        Popular Movies
      </button>
      <input
        name="movie"
        onChange={[Function]}
        placeholder="enter movie name"
        type="text"
        value=""
      />
    </div>
    <div
      className="all-movies"
    />
  </div>
</div>
`;

exports[`<App/> should not defer from snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="App"
      >
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/detail/:id"
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/detail/:id"
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/detail/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="App"
        >
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/detail/:id"
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/detail/:id"
            />,
          ],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/detail/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<app/> create  snapshot 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="nav-bar"
    >
      <h1>
         Welcome to Movies guide 
      </h1>
      <button
        onClick={[Function]}
      >
        All Movies
      </button>
      <button
        onClick={[Function]}
      >
        Top Movies
      </button>
      <button
        onClick={[Function]}
      >
        Playing Movies
      </button>
      <button
        onClick={[Function]}
      >
        Popular Movies
      </button>
      <input
        name="movie"
        onChange={[Function]}
        placeholder="enter movie name"
        type="text"
        value=""
      />
    </div>
    <div
      className="all-movies"
    />
  </div>
</div>
`;

exports[`<app/> should not defer from snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="App"
      >
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/detail/:id"
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/detail/:id"
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/detail/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="App"
        >
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/detail/:id"
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/detail/:id"
            />,
          ],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/detail/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
